src/components/swap/SwapRoute.tsx

```diff
@@ -1,64 +1,83 @@
+function getGasPrice(trade: ClassicTrade) {
+  const chainId = trade.inputAmount.currency.chainId
+  if (!trade.gasUseEstimateUSD || !SUPPORTED_GAS_ESTIMATE_CHAIN_IDS.includes(chainId)) return undefined
 
-  const routes = getRoutingDiagramEntries(trade)
+  return trade.gasUseEstimateUSD === 0 ? '<$0.01' : '$' + trade.gasUseEstimateUSD.toFixed(2)
+}
 
-  const gasPrice =
-    // TODO(WEB-2022)
-    // Can `trade.gasUseEstimateUSD` be defined when `chainId` is not in `SUPPORTED_GAS_ESTIMATE_CHAIN_IDS`?
-    trade.gasUseEstimateUSD && chainId && SUPPORTED_GAS_ESTIMATE_CHAIN_IDS.includes(chainId)
-      ? trade.gasUseEstimateUSD === 0
-        ? '<$0.01'
-        : '$' + trade.gasUseEstimateUSD.toFixed(2)
-      : undefined
+function RouteLabel({ trade }: { trade: SubmittableTrade }) {
+  return (
+    <RowBetween>
+      <ThemedText.BodySmall color="neutral2">Order Routing</ThemedText.BodySmall>
+      <RouterLabel trade={trade} color="neutral1" />
+    </RowBetween>
+  )
+}
 
+function PriceImpactRow({ trade }: { trade: ClassicTrade }) {
+  const { formatPriceImpact } = useFormatter()
   return (
+    <ThemedText.BodySmall color="neutral2">
+      <RowBetween>
+        <Trans>Price Impact</Trans>
+        <div>{formatPriceImpact(trade.priceImpact)}</div>
+      </RowBetween>
+    </ThemedText.BodySmall>
+  )
+}
+
+export function RoutingTooltip({ trade }: { trade: SubmittableTrade }) {
+  return isClassicTrade(trade) ? (
+    <Column gap="md">
+      <PriceImpactRow trade={trade} />
+      <Separator />
+      <RouteLabel trade={trade} />
+      <SwapRoute trade={trade} />
+    </Column>
+  ) : (
     <Column gap="md">
-      <RouterLabel trade={trade} color="neutral2" />
+      <RouteLabel trade={trade} />
       <Separator />
-      {syncing ? (
-        <LoadingRows>
-          <div style={{ width: '100%', height: '30px' }} />
-        </LoadingRows>
-      ) : (
-        <RoutingDiagram
-          currencyIn={trade.inputAmount.currency}
-          currencyOut={trade.outputAmount.currency}
-          routes={routes}
-        />
-      )}
-      {autoRouterSupported && (
-        <>
-          <Separator />
-          {syncing ? (
-            <LoadingRows>
-              <div style={{ width: '100%', height: '15px' }} />
-            </LoadingRows>
-          ) : (
-            <ThemedText.Caption color="neutral2">
-              {gasPrice ? <Trans>Best price route costs ~{gasPrice} in gas.</Trans> : null}{' '}
-              <Trans>
-                This route optimizes your total output by considering split routes, multiple hops, and the gas cost of
-                each step.
-              </Trans>
-            </ThemedText.Caption>
-          )}
-        </>
-      )}
+      <UniswapXDescription />
+    </Column>
+  )
+}
+
+export function SwapRoute({ trade }: { trade: ClassicTrade }) {
+  const { inputAmount, outputAmount } = trade
+  const routes = getRoutingDiagramEntries(trade)
+  const gasPrice = getGasPrice(trade)
```

Comment:
you can do the following to get the same formatting if you return a number in the `getGasPrice` function:

```
const gasPrice = useFormatter().formatNumber({ input: getGasePrice(trade), type: [
  { exact: 0, hardCodedInput: { input: 0.01, prefix: '<' }, formatterOptions: {
    notation: 'standard',
    maximumFractionDigits: 2,
    minimumFractionDigits: 2,
  }, {
    upperBound: Infinity, formatterOptions: {
    notation: 'standard',
    maximumFractionDigits: 2,
    minimumFractionDigits: 2,
    },
  }
] 
})
```

Reply:
wow that looks terrifying. yeah I think updating NumberType.FiatGasPrice to include the exact zero case in sep PR probably makes sense